test/functional/ex_cmds/lsp_spec.lua

```diff
@@ -0,0 +1,150 @@
+local t = require('test.testutil')
+local n = require('test.functional.testnvim')()
+local t_lsp = require('test.functional.plugin.lsp.testutil')
+
+local clear = n.clear
+local eq = t.eq
+local exec_lua = n.exec_lua
+
+local create_server_definition = t_lsp.create_server_definition
+
+describe(':lsp', function()
+  before_each(function()
+    clear()
+    exec_lua(create_server_definition)
+    exec_lua(function()
+      local server = _G._create_server()
+      vim.lsp.config('dummy', {
+        filetypes = { 'lua' },
+        cmd = server.cmd,
+      })
+      vim.cmd('set ft=lua')
+    end)
+  end)
+
+  it('enable with arguments', function()
```

Comment:
Also need 1 mini-test which just checks how things fail with `nvim -u NONE` (i.e. when the runtime is missing). Mainly just want to check that nvim doesn't crash.

```lua
    clear {
      args_rm = { '-u' },
      args = { '-u', 'NONE' },
      env = { VIMRUNTIME = 'non-existent' },
    }
    eq('..', t.pcall_err(n.command, 'lsp enable dummy'))
```